<?xml version="1.0"?>
<launch>
  <arg name="input_pointcloud_topic" default="/arm_cam3d/depth_registered/points" />
  <arg name="camera_frame" default="arm_cam3d_rgb_optical_frame" />
 

  <!-- This launch file brings up the complete scene segmentation pipeline with reasonable (for @Work) parameters. -->

  <arg name="workspace_finder_config_file" default="PASS THE CONFIGFILE NAME AS ARGUMENT" />

  <include file="$(find mcr_scene_segmentation)/ros/launch/workspace_finder.launch">
    <arg name="input_pointcloud_topic" value="$(arg input_pointcloud_topic)" />
  </include>

  <group ns="mcr_perception">

    <node pkg="mcr_scene_segmentation" type="workspace_finder" name="workspace_finder" output="screen">
      <remap from="~input/polygon" to="/mcr_perception/convex_hull/output_polygon" />
      <remap from="~input/coefficients" to="/mcr_perception/planar_segmentation/model" />
      <remap from="~workspace_polygon" to="/visualization_marker" />
    </node>

    <node pkg="mcr_scene_segmentation" type="tabletop_cloud_accumulator" name="tabletop_cloud_accumulator" output="screen">
      <param name="accumulate_clouds" value="5" type="int"/>
      <param name="accumulation_timeout" value="10" type="int"/>
      <param name="octree_resolution" value="0.0025" type="double"/>
      <param name="min_height" value="0.01" type="double"/>
      <param name="max_height" value="0.10" type="double"/>
      <param name="camera_frame" value="$(arg camera_frame)" />
      <remap from="~input_pointcloud" to="/mcr_perception/transform/output" />
    </node>

    <node pkg="mcr_scene_segmentation" type="tabletop_cloud_clusterer" name="tabletop_cloud_clusterer" output="screen" >
       <param name="outlier_radius" value="0.03" type="double" />
       <param name="cluster_tolerance" value="0.02" type="double" />
       <param name="min_distance_to_polygon" value="0.03" type="double" />
    </node>

    <node pkg="mcr_scene_segmentation" type="bounding_box_maker" name="bounding_box_maker" output="screen" >
      <remap from="~bounding_boxes" to="/visualization_marker" />
    </node>
  
  </group>

</launch>
